### PR TITLE
feat: saga pattern for consultant creation

### PR DESCRIPTION
**Pull request by tkuzynow**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

**Review by Leandro13Silva13 — DISMISSED** [review]

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java [hunk 1]
```diff
@@ -55,19 +61,25 @@ public class ConsultantCreatorService {
...
+
   /**
    * Creates a new {@link Consultant} by {@link CreateConsultantDTO} in database, keycloak and
    * rocket chat.
    *
    * @param createConsultantDTO the input used for creation
    * @return the generated {@link Consultant}
    */
-  public Consultant createNewConsultant(CreateConsultantDTO createConsultantDTO) {
+  private Consultant createNewConsultantWithoutAppointment(
```
Comment (patric-dosch-vi) [inline src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java]: as I see that was already ... but I would comment the public method, not the private. And move the private under the public.

Comment (patric-dosch-vi) [reply src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java]: and maybe extend the comment with the appointment service (optional). I think that's a valid comment.

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java [hunk 2]
```diff
@@ -81,6 +93,46 @@ public Consultant createNewConsultant(CreateConsultantDTO createConsultantDTO) {
...
+      log.error(
+          "User with id {}, who has roles {}, has created a consultant with id {} but the appointment service returned an error: {}",
+          authenticatedUser.getUserId(),
+          authenticatedUser.getRoles(),
+          newConsultant.getId(),
+          e.getMessage());
+      this.rollbackCreateNewConsultant(newConsultant);
+      throw new DistributedTransactionException(
+          e,
+          new DistributedTransactionInfo(
```
Comment (patric-dosch-vi) [inline src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java]: also use the builder? makes it easy to read

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/ConsultantAdminService.java [hunk 3]
```diff
@@ -90,36 +89,7 @@ private static String getDisplayNameFromUserMap(Map<String, Object> map) {
...
-      throw new DistributedTransactionException(
-          e,
-          new DistributedTransactionInfo(
-              "createNewConsultant",
-              completedSteps,
-              TransactionalStep.CREATE_ACCOUNT_IN_CALCOM_OR_APPOINTMENTSERVICE));
-    }
-
-    return consultantAdminResponseDTO;
+    return this.consultantCreateSaga.createNewConsultant(createConsultantDTO);
```
Comment (patric-dosch-vi) [inline src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/ConsultantAdminService.java]: do you like that this.? I dont... :) 
it's also not in all places, e.g. userHelper.getRandomPassword() but this is not important for now. it's just a general question

**Review by patric-dosch-vi — COMMENTED** [review]
hey @tkuzynow, looks all fine. Just some minor comments. I would fix the comment, the builder.
And for sure you could fix the sonar-issue. Is that also in the UpdateSaga that is already implemented? Then I would also use a constant there.

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/ConsultantAdminService.java (hunk 3 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/ConsultantAdminService.java]: I agree, it's redundant, I removed it

File: src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java (hunk 1 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/ConsultantCreateSaga.java]: done

**Review by patric-dosch-vi — APPROVED** [review]
hey @tkuzynow, looks very good. Thanks.

**Pull request merged**